aspects/aspects.go

```diff
@@ -311,26 +329,60 @@ type Aspect struct {
 // Set sets the named aspect to a specified value.
 func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 	if err := validateAspectDottedPath(request, nil); err != nil {
-		return fmt.Errorf(`cannot parse Set request: %w`, err)
+		return badRequestErrorFrom(a, "set", request, err.Error())
 	}
 
-	requestSubkeys := strings.Split(request, ".")
+	var matches []requestMatch
+	subkeys := strings.Split(request, ".")
 	for _, accessPatt := range a.accessPatterns {
-		placeholders, _, ok := accessPatt.match(requestSubkeys)
+		placeholders, suffixParts, ok := accessPatt.match(subkeys)
 		if !ok {
 			continue
 		}
 
-		storagePath, err := accessPatt.storagePath(placeholders)
-		if err != nil {
-			return err
+		for _, part := range suffixParts {
+			if len(part) > 1 && part[0] == '{' && part[len(part)-1] == '}' {
+				return badRequestErrorFrom(a, "set", request, "cannot set with unmatched placeholders")
```

Comment:
do we want a TODO here? ultimately we want to support this

Reply:
Sure, I'll add one